tests/e2e/csi_snapshot_utils.go

```diff
@@ -698,35 +703,37 @@ func waitForVolumeSnapshotContentReadyToUse(client snapclient.Clientset, ctx con
 	var volumeSnapshotContent *snapV1.VolumeSnapshotContent
 	var err error
 
-	waitErr := wait.PollImmediate(poll, pollTimeout*2, func() (bool, error) {
-		volumeSnapshotContent, err = client.SnapshotV1().VolumeSnapshotContents().Get(ctx, name, metav1.GetOptions{})
-		framework.Logf("volumesnapshotcontent details: %v", volumeSnapshotContent)
-		if err != nil {
-			return false, fmt.Errorf("error fetching volumesnapshotcontent details : %v", err)
-		}
-		if volumeSnapshotContent.Status != nil && *volumeSnapshotContent.Status.ReadyToUse {
-			framework.Logf("%s volume snapshotContent is in ready state", name)
-			return true, nil
-		}
-		return false, nil
-	})
+	waitErr := wait.PollUntilContextTimeout(ctx, poll, pollTimeout*2, true,
```

Comment:
just curious, why are we changing `PollImmediate()` to `PollUntilContextTimeout()`